Apps/Examples/Examples/All Examples/AnimateImageLayerExample.swift

```diff
@@ -0,0 +1,103 @@
+import MapboxMaps
+
+@objc(AnimateImageLayerExample)
+class AnimateImageLayerExample: UIViewController, ExampleProtocol {
+    var mapView: MapView!
+    var sourceId = "radar-source"
+    var timer: Timer?
+    var imageNumber = 0
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        let center = CLLocationCoordinate2D(latitude: 41.874, longitude: -75.789)
+        let cameraOptions = CameraOptions(center: center, zoom: 5)
+        let mapInitOptions = MapInitOptions(cameraOptions: cameraOptions, styleURI: .dark)
+        mapView = MapView(frame: view.bounds, mapInitOptions: mapInitOptions)
+        mapView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+
+        // Hide the `scaleBar` at all zoom levels.
+        mapView.ornaments.options.scaleBar.visibility = .hidden
+
+        // This also updates the color of the info button to match the map's style.
+        mapView.tintColor = .lightGray
+
+        // Set the map's `CameraBoundsOptions` to limit the map's zoom level.
+        mapView.camera.options.maxZoom = 5.99
+        mapView.camera.options.minZoom = 4
+
+        view.addSubview(mapView)
+
+        mapView.mapboxMap.onNext(.mapLoaded) { _ in
+            self.addImageLayer()
+        }
+    }
+
+    func addImageLayer() {
+        let style = mapView.mapboxMap.style
+
+        // Create an `ImageSource`. This will manage the image displayed in the `RasterLayer` as well
+        // as the location of that image on the map.
+        var imageSource = ImageSource()
+
+        // Set the `coordinates` property to an array of longitude, latitude pairs.
+        imageSource.coordinates = [
+            [-80.425, 46.437],
+            [-71.516, 46.437],
+            [-71.516, 37.936],
+            [-80.425, 37.936]
+        ]
+
+        // Get the file path for the first radar image, then set the `url` for the `ImageSource` to that path.
+        let path = Bundle.main.path(forResource: "radar0", ofType: "gif")!
+        imageSource.url = path
+
+        // Create a `RasterLayer` that will display the images from the `ImageSource`
+        var imageLayer = RasterLayer(id: "radar-layer")
+        imageLayer.source = sourceId
+
+        // Set `rasterFadeDuration` to `0`. This prevents visible transitions when the image is updated.
+        imageLayer.rasterFadeDuration = .constant(0)
+
+        do {
+            try style.addSource(imageSource, id: sourceId)
+            try style.addLayer(imageLayer)
+
+            // Add a tap gesture recognizer that will allow the animation to be stopped and started.
+            let tapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(manageTimer))
+            mapView.addGestureRecognizer(tapGestureRecognizer)
+        } catch {
+            print("Failed to add the source or layer to style. Error: \(error)")
+        }
+        manageTimer()
+    }
+
+    @objc func manageTimer() {
```

Comment:
why is this method `@objc`?

Reply:
it is also the [action](https://github.com/mapbox/mapbox-maps-ios/pull/498/files/1449203817508cd6a82627e09bebc0960d503193#diff-575f459c303712f912f88fb0cc0918fe6a34693e859ee6e0f6ec1939cf6a4fb3R67) for the tap gesture recognizer